src/main/java/com/bobocode/hoverla/bring/context/ApplicationContextImpl.java

```diff
@@ -1,60 +1,152 @@
-           4) Convert bean list to map
-           5) Send beans map to initializer to instantiate beans and inject dependencies
-         */
+
+        List<BeanDefinition> beanDefinitionList = scanningApplication(scanners);
+        validate(validator, beanDefinitionList);
+        contextCreation(beanDefinitionList);
+        beansInstantiation(initializer);
+
+        log.info("Application context initialized");
+    }
+
+    private List<BeanDefinition> scanningApplication(List<BeanScanner> scanners) {
+        List<BeanDefinition> beanDefinitionList = new ArrayList<>();
+        log.info("Scanning application for finding bean definitions");
+        for (BeanScanner beanScanner : scanners) {
+            log.debug("Running bean scanner {}", beanScanner.getClass().getSimpleName());
+            beanDefinitionList.addAll(beanScanner.scan());
+        }
+        return beanDefinitionList;
+    }
+
+    private void validate(BeanValidator validator, List<BeanDefinition> beanDefinitionList) {
+        log.info("Beans definitions validating");
+        validator.validate(beanDefinitionList);
+    }
+
+    private void contextCreation(List<BeanDefinition> beanDefinitionList) {
+        log.debug("Context table generation");
+        for (BeanDefinition beanDefinition : beanDefinitionList) {
+            log.debug("Bean definition with name = {} and type = {} will be added", beanDefinition.name(), beanDefinition.type());
+            context.put(beanDefinition.name(), beanDefinition.type(), beanDefinition);
+        }
+    }
+
+    private void beansInstantiation(BeanInitializer initializer) {
+        log.info("Beans initialization");
+        initializer.initialize(context);
     }
 
-    @Override
-    public <T> void register(String name, Class<T> beanType) {
 
+    @Override
+    public <T> void register(String beanName, Class<T> beanType) {
+        // TODO:  add implementation
     }
 
     @Override
     public void register(Class<?>... beanTypes) {
-
+        // TODO:  add implementation
     }
 
     @Override
     public <T> String register(Class<T> beanType) {
         return null;
+        // TODO: add implementation
     }
 
     @Override
     public <T> T getBean(Class<T> beanType) {
-        return null;
+        Objects.requireNonNull(beanType, BEAN_TYPE_MUST_BE_NOT_NULL_MESSAGE);
+
+        List<T> beans = context.column(beanType)
+                .values()
+                .stream()
+                .map(beanDefinition -> beanType.cast(beanDefinition.instance()))
+                .toList();
+        if (beans.isEmpty()) {
+            throw new NoSuchBeanException(NO_SUCH_BEAN_EXCEPTION_MESSAGE.formatted(beanType.getSimpleName()));
+        } else if (beans.size() > 1) {
+            throw new NoUniqueBeanException("Expected single bean of type " + beanType.getSimpleName()
+                + ", but found " + beans.size());
+        } else return beans.get(0);
     }
 
     @Override
     public Object getBean(String beanName) {
```

Comment:
We should check here that `beanName` is not null, empty or blank

Reply:
I agree with you about null check. But we discussed at the meeting that empty or blank name is acceptable.

Reply:
@nadya-gainutsa uh, okay, will discuss it with others...

Reply:
@DanielChanger do you mean that beanName should not contains spaces at all? Or it can be name with spaces but not blank?

Reply:
I found the answer below :)

Reply:
Done